### PR TITLE
system.go: Check for existense of lsof command

### PR DESCRIPTION
**Pull request by sladyn98**
Added function to check availablity of
a particualr command.
Update the tests to factor in the return 0.

### CODE REVIEW

File: system.go [hunk 1]
```diff
@@ -11,12 +11,24 @@ import (
 )
 
 func countOpenFiles() int {
-	out, err := exec.Command("/bin/sh", "-c", fmt.Sprintf("lsof -p %v", os.Getpid())).Output()
-	if err != nil {
-		log.Fatal(err)
+
```
Comment (glaslos) [inline system.go]:
```suggestion
```
You have to buy me a coffee soon every time you have this extra line in a PR :wink:

File: system.go [hunk 2]
```diff
@@ -11,12 +11,24 @@ import (
...
+	if runtime.GOOS == "linux" {
+
```
Comment (glaslos) [inline system.go]:
```suggestion
```

File: system.go [hunk 3]
```diff
@@ -11,12 +11,24 @@ import (
...
+			out, err := exec.Command("/bin/sh", "-c", fmt.Sprintf("lsof -p %d", os.Getpid())).Output()
+			if err != nil {
+				log.Fatal(err)
+			}
+
+			lines := strings.Split(string(out), "\n")
+			return len(lines) - 1
+		}
+
+		log.Fatalln("lsof command does not exist.kindly run sudo-apt-get install lsof")
```
Comment (glaslos) [inline system.go]:
```suggestion
		log.Fatalln("lsof command does not exist. Kindly run sudo apt install lsof")
```

File: system.go [hunk 4]
```diff
@@ -11,12 +11,24 @@ import (
...
+
 	}
-	lines := strings.Split(string(out), "\n")
-	return len(lines) - 1
+	log.Fatalln("this command is not available on non-linux based operating systems")
```
Comment (glaslos) [inline system.go]: So Glutton doesn't run on any non Linux OS?

File: system.go [hunk 5]
```diff
@@ -43,3 +55,12 @@ func (g *Glutton) startMonitor(quit chan struct{}) {
 		}
 	}()
 }
+
+func isCommandAvailable(name string) bool {
+
+	cmd := exec.Command("/bin/sh", "-c", "command -v "+name)
+	if err := cmd.Run(); err != nil {
```
Comment (glaslos) [inline system.go]: Did you test that this returns an error for an unknown command?

Comment (sladyn98) [reply system.go]: Yep tried and tested boss:)

File: system.go (hunk 4 above)
Comment (sladyn98) [reply system.go]: You said glutton was only for Linux :)

File: system.go (hunk 1 above)
Comment (sladyn98) [reply system.go]: Doesn't go fmt take care of this.i hope you spent less time reviewing this commit :)

File: system.go (hunk 4 above)
Comment (glaslos) [reply system.go]: But then what's the point of `runtime.GOOS == "linux"` ?

Comment (sladyn98) [reply system.go]: Yeah I agree with you we could then make the message "command not available" regardless.?

Comment (glaslos) [reply system.go]: How about let the function return an error and not report the number of open files on error?

Comment (sladyn98) [reply system.go]: I do not think that would help much, we could rather do a `log.Fatal` which would anyways cause a `sys.exit()` and would exit the function and if we even return an error we would by default have to return some value for int, which in that case would be 0.

**Comment by glaslos** [comment]
I don't like the fact that this function can terminate Glutton. The function should never terminate Glutton but return an appropriate error.

**Comment by sladyn98** [comment]
@glaslos  I did make the changes and returned an int as well as an error if the function fails.I modified the tests as well , just have a look.I am not quite sure if you want the `test_function` to test for errors.But do have a look and review.Thanks

File: system.go [hunk 6]
```diff
@@ -43,3 +53,11 @@ func (g *Glutton) startMonitor(quit chan struct{}) {
 		}
 	}()
 }
+
+func isCommandAvailable(name string) bool {
+	cmd := exec.Command("/bin/sh", "-c", "command -v "+name)
```
Comment (glaslos) [inline system.go]: I think this line needs gofmt

File: system_test.go [hunk 7]
```diff
@@ -5,8 +5,13 @@ import (
 )
 
 func TestCountOpenFiles(t *testing.T) {
-	openFiles := countOpenFiles()
-	if openFiles <= 0 {
+	openFiles, err := countOpenFiles()
+
+	if err!=nil{
```
Comment (glaslos) [inline system_test.go]: gofmt

File: system.go (hunk 6 above)
Comment (sladyn98) [reply system.go]: @glaslos I already applied gofmt on the command line :) This line still remained unmodified

**Pull request merged**